### PR TITLE
[main] [feat] smoke test with foundry fork test

### PR DESCRIPTION
**Pull request by john-xina-p88**

### CODE REVIEW

File: test/fork/smoke-test/Smoke_MaxProfit.t.sol [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    bytes32[] memory positionIds = ForkEnv.positionReader.getForceTakeMaxProfitablePositionIds(
+      10,
+      0,
+      prices,
+      shouldInverts
+    );
+
+    if (positionIds.length == 0) {
+      console.log("No position to be deleveraged");
```
Comment (kim-hera) [inline test/fork/smoke-test/Smoke_MaxProfit.t.sol:32]: Misleading description

File: test/fork/smoke-test/Smoke_MaxProfit.t.sol [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    for (uint i = 0; i < positionIds.length; i++) {
+      IPerpStorage.Position memory _position = ForkEnv.perpStorage.getPositionById(positionIds[i]);
+      if (
+        _position.primaryAccount == address(0) ||
+        _checkIsUnderMMR(
+          _position.primaryAccount,
+          _position.subAccountId,
+          _position.marketIndex,
+          _position.avgEntryPriceE30
+        )
```
Comment (kim-hera) [inline test/fork/smoke-test/Smoke_MaxProfit.t.sol:55]: Can we remove this check, or any other workaround? There is a chance that every account could justgo  through this check without performing `forceTakeMaxProfit`.

Comment (john-xina-p88) [reply test/fork/smoke-test/Smoke_MaxProfit.t.sol:55]: We have that check in `Deleverage` will follow that to make sure at least there is one force close here.

File: test/fork/smoke-test/Smoke_Liquidity.t.sol [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    (
+      uint256 _minPublishTime,
+      bytes32[] memory _priceUpdateCalldata,
+      bytes32[] memory _publishTimeUpdateCalldata
+    ) = ForkEnv.ecoPythBuilder.build(data);
+
+    // hlp price = aum / total supply
+    uint256 _hlpPriceE30 = (ForkEnv.calculator.getAUME30(false) * 1e18) / ForkEnv.hlp.totalSupply();
+    uint256 _estimatedHlpReceived = (10 * 1e18 * 1e30) / _hlpPriceE30;
+
```
Comment (kim-hera) [inline test/fork/smoke-test/Smoke_Liquidity.t.sol:59]: Add some block warp, time warp between order creation and execution for more realisticness.
Say 15 sec and 30 blocks.

File: test/fork/smoke-test/Smoke_Trade.t.sol [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+      _orderIndex = ForkEnv.limitTradeHandler.limitOrdersIndex(subAccount);
+    }
+
+    IEcoPythCalldataBuilder.BuildData[] memory data = _buildDataForPrice();
+    (
+      uint256 _minPublishTime,
+      bytes32[] memory _priceUpdateCalldata,
+      bytes32[] memory _publishTimeUpdateCalldata
+    ) = ForkEnv.ecoPythBuilder.build(data);
+
```
Comment (kim-hera) [inline test/fork/smoke-test/Smoke_Trade.t.sol:83]: Add some warp, like the one mention previously on Smoke_Liquidity

File: test/fork/smoke-test/Smoke_TriggerOrder.t.sol [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+    tokenIndexes[20] = 0x424e420000000000000000000000000000000000000000000000000000000000;
+    tokenIndexes[21] = 0x534f4c0000000000000000000000000000000000000000000000000000000000;
+    tokenIndexes[22] = 0x5151510000000000000000000000000000000000000000000000000000000000;
+    tokenIndexes[23] = 0x5852500000000000000000000000000000000000000000000000000000000000;
+    tokenIndexes[24] = 0x4e56444100000000000000000000000000000000000000000000000000000000;
+    tokenIndexes[25] = 0x4c494e4b00000000000000000000000000000000000000000000000000000000;
+    tokenIndexes[26] = 0x4348460000000000000000000000000000000000000000000000000000000000;
+    tokenIndexes[27] = 0x444f474500000000000000000000000000000000000000000000000000000000;
+    tokenIndexes[28] = 0x4341440000000000000000000000000000000000000000000000000000000000;
+    tokenIndexes[29] = 0x5347440000000000000000000000000000000000000000000000000000000000;
```
Comment (kim-hera) [inline test/fork/smoke-test/Smoke_TriggerOrder.t.sol:198]: Derive from config storage might be more convenient?

File: test/fork/smoke-test/Smoke_TriggerOrder.t.sol [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+      data[i - 1].maxDiffBps = 15_000;
+    }
+  }
+
+  function _buildPrice_Trigger(
+    uint256 _marketIndex,
+    uint256 _triggerPrice,
+    bool _above
+  ) internal view returns (uint64[] memory prices) {
+    bytes32[] memory tokenIndexes = new bytes32[](30);
```
Comment (kim-hera) [inline test/fork/smoke-test/Smoke_TriggerOrder.t.sol:166]: It's not token.

File: test/fork/smoke-test/Smoke_TriggerOrder.t.sol [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+  address[] internal executeAccounts;
+  uint8[] internal executeSubAccountIds;
+  uint256[] internal executeOrderIndexes;
+
+  function setUp() public virtual override {
+    super.setUp();
+  }
+
+  function testCheck_SmokeTrigger() external view {
+    console2.logBytes(abi.encodeWithSignature("ITradeService_InsufficientFreeCollateral()"));
```
Comment (kim-hera) [inline test/fork/smoke-test/Smoke_TriggerOrder.t.sol:36]: How does this work? Or just debug playground/

**Review by kim-hera — APPROVED** [review]

**Pull request merged**